apps/maestro/src/features/InterchainTokenDeployment/hooks/useDeployAndRegisterRemoteInterchainTokenMutation.ts

```
@@ -160,18 +201,74 @@ const useMulticallArgs = ({
+      if (!value18Decimals) {
+        return 0n;
+      }
+
+      const chainGasValueDecimals = CHAIN_GAS_VALUE_DECIMALS[chainId];
+
+      if (!chainGasValueDecimals) {
+        return value18Decimals;
+      }
+
+      const targetDecimals = 18;
+      const factor = 10n ** BigInt(targetDecimals - chainGasValueDecimals);
+
+      return value18Decimals / factor;
+    };
+
     const registerTxData = destinationChainIds.map((destinationChain, i) =>
       INTERCHAIN_TOKEN_FACTORY_ENCODERS.deployRemoteInterchainToken.data({
         ...commonArgs,
         destinationChain,
-        gasValue: input.remoteDeploymentGasFees?.gasFees?.[i].fee ?? 0n,
+        gasValue: scaleGasValue(
+          input.remoteDeploymentGasFees?.gasFees?.[i].fee
+        ),
       })
     );
 
     return [deployTxData, ...registerTxData];
   }, [input, tokenId, destinationChainIds, chainId]);
 
-  return multicallArgs;
+  const totalGasFee = input?.remoteDeploymentGasFees?.totalGasFee ?? 0n;
+
+  const [isTokenReadyForMulticall, setIsTokenReadyForMulticall] =
+    useState(false);
+
+  useEffect(() => {
+    setIsTokenReadyForMulticall(
+      !CHAIN_IDS_WITH_MANUAL_MULTICALL.includes(chainId)
+    );
+  }, [chainId]);
+
+  // enable if there are no remote chains or if there are remote chains and the total gas fee is greater than 0
+  const isDestinationFeeSet = !destinationChainIds.length || totalGasFee > 0n;
+  const isMutationReady =
+    multicallArgs.length > 0 && isDestinationFeeSet && isTokenReadyForMulticall;
+
+  const { data: prepareMulticall, error: simulationError } =
+    useSimulateInterchainTokenFactoryMulticall({
+      chainId,
+      value: totalGasFee,
+      args: [multicallArgs],
+      query: {
+        enabled: isMutationReady,
+      },
+    });
+
+  if (simulationError) {
+    console.error(
+      "useDeployAndRegisterRemoteInterchainTokenMutation simulation error:",
+      simulationError
+    );
+  }
```

This might result in silent failures. Shouldn't we surface this error so it can maybe be passed to the user?

---

I agree, as it provides a stuck UI, but I don't think that it's in the scope of this PR. What do you say?

---

Hmm could you elaborate on how propagating the error would work?

---

@joaoazevedo Since the error was also not handled properly before, let's fix this in a new PR. Added this task on Linear: https://linear.app/common-prefix/issue/ITS-81